app/handler/trades.js

```diff
@@ -73,6 +77,395 @@ exports.getActiveOffer = function (steamID) {
+    }
+
+    if (Object.getOwnPropertyNames(cart[partner]['itemsToReceive']).length === 0) {
+        // We are not taking any items, don't request their inventory
+
+        alteredMessage = createAlteredMessage(alteredItems, partner);
+
+        if (Object.getOwnPropertyNames(cart[partner]['itemsToGive']).length === 0) {
+            exports.removeFromCart(true, partner);
+            callback(null, alteredMessage);
+            return;
+        }
+
+        offer.data('dict', itemsDict);
+        offer.data('diff', itemsDiff);
+
+        offer.data('handleTimestamp', start);
+
+        offer.setMessage('Powered by TF2 Automatic');
+
+        client.chatMessage(partner, (alteredMessage || 'Please wait while I process your offer...'));
+
+        require('app/trade').sendOffer(offer, function (err) {
+            if (err) {
+                if (err.message.indexOf('We were unable to contact the game\'s item server') !== -1) {
+                    return callback(null, 'Team Fortress 2\'s item server may be down or Steam may be experiencing temporary connectivity issues');
+                } else if (err.message.indexOf('can only be sent to friends') != -1) {
+                    return callback(err);
+                } else if (err.message.indexOf('maximum number of items allowed in your Team Fortress 2 inventory') > -1) {
+                    return callback(null, 'I don\'t have space for more items in my inventory');
+                } else if (err.eresult !== undefined) {
+                    if (err.eresult == 10) {
+                        callback(null, 'An error occurred while sending your trade offer, this is most likely because I\'ve recently accepted a big offer');
+                    } else if (err.eresult == 15) {
+                        callback(null, 'I don\'t, or you don\'t, have space for more items');
+                    } else if (err.eresult == 16) {
+                        // This happens when Steam is already handling an offer (usually big offers), the offer should be made
+                        callback(null, 'An error occurred while sending your trade offer, this is most likely because I\'ve recently accepted a big offer');
+                    } else if (err.eresult == 20) {
+                        callback(null, 'Team Fortress 2\'s item server may be down or Steam may be experiencing temporary connectivity issues');
+                    } else {
+                        callback(null, 'An error occurred while sending the offer (' + TradeOfferManager.EResult[err.eresult] + ')');
+                    }
+                    return;
+                }
+            }
+
+            return callback(err);
+        });
+
+        return;
+    }
+
+    inventory.getDictionary(partner, false, function (err, theirDict) {
+        if (err) {
+            return callback(null, 'Failed to load inventories, Steam might be down');
+        }
+
+        for (const name in cart[partner]['itemsToReceive']) {
```

Comment:
Please don't do `['itemsToReceive']`, just do `cart[partner].itemsToReceive`. itemsToGive should also be changed to their so that it matches how the buy / sell command works.